File: cstartup/src/main/java/com/eju/cstartup/StartupException.java
package com.eju.cstartup;

import androidx.annotation.NonNull;
import androidx.annotation.RestrictTo;

/**
 * The Runtime Exception thrown by the android.startup library.
 *
 * @hide
 */
@RestrictTo(RestrictTo.Scope.LIBRARY)
@SuppressWarnings("WeakerAccess")
public final class StartupException extends RuntimeException {
    public StartupException(@NonNull String message) {
        super(message);
    }

    public StartupException(@NonNull Throwable throwable) {
        super(throwable);
    }

    public StartupException(@NonNull String message, @NonNull Throwable throwable) {
        super(message, throwable);
    }
}

File: startup/src/main/java/com/eju/startup/StartupException.kt
package com.eju.startup

import androidx.annotation.RestrictTo
import java.lang.RuntimeException

/**
 * The Runtime Exception thrown by the android.startup library.
 *
 * @hide
 */
@RestrictTo(RestrictTo.Scope.LIBRARY)
class StartupException : RuntimeException {
    constructor(message: String) : super(message) {}
    constructor(throwable: Throwable) : super(throwable) {}
    constructor(message: String, throwable: Throwable) : super(message, throwable) {}
}